src/BlazorWasmSdk/Tasks/ComputeBlazorBuildAssets.cs

```diff
@@ -97,7 +100,16 @@ public override bool Execute()
                     }
 
                     var destinationSubPath = candidate.GetMetadata("DestinationSubPath");
-                    if (string.IsNullOrEmpty(destinationSubPath))
+                    if (candidate.GetMetadata("FileName") == "dotnet" && candidate.GetMetadata("Extension") == ".js")
+                    {
+                        var dotNetVersion = MicrosoftNETCoreAppRefPackageVersion;
+                        var itemHash = FileHasher.GetFileHash(candidate.ItemSpec);
+                        var cacheBustedDotnetJSFileName = $"dotnet.{dotNetVersion}.{itemHash}";
+                        candidate.SetMetadata("RelativePath", $"_framework/{cacheBustedDotnetJSFileName}.js");
+                        candidate.SetMetadata("AssetTraitName", "BlazorWebAssemblyResource");
+                        candidate.SetMetadata("AssetTraitValue", "native");
+                    }
+                    else if (string.IsNullOrEmpty(destinationSubPath))
```

Comment:
I'm not 100% sure you should be editing the candidate (don't remember the exact details) but instead creating a new static web asset from scratch (the candidate is not a static web asset, it's a "random thing with properties")

We might also want to do this through StaticWebAsset.cs as that applies a lot of normalization to the item properties.

Reply:
> creating a new static web asset from scratch (the candidate is not a static web asset, it's a "random thing with properties")

In that case where would we "put" the newly created static web asset. I ask as the `AssetCandidates` is a collection of `ITaskItem` and thus won't accept a `StaticWebAsset`. Maybe we should be doing something like:

https://github.com/dotnet/sdk/blob/a0d74e39bc7533643b78184d189336ca2b96b2be/src/BlazorWasmSdk/Tasks/ComputeBlazorBuildAssets.cs#L87-L96

It ensures we don't update the initial asset type, but still are able to add in the desired properties. 

> We might also want to do this through StaticWebAsset.cs as that applies a lot of normalization to the item properties.

Can leverage:
https://github.com/dotnet/sdk/blob/a0d74e39bc7533643b78184d189336ca2b96b2be/src/RazorSdk/Tasks/StaticWebAssets/StaticWebAsset.cs#L46-L54